diesel/src/macros/mod.rs

```diff
@@ -19,9 +19,10 @@ macro_rules! __diesel_column {
 
         impl<DB> $crate::query_builder::QueryFragment<DB> for $column_name where
             DB: $crate::backend::Backend,
+            <$($table)::* as QuerySource>::FromClause: QueryFragment<DB>,
         {
             fn to_sql(&self, out: &mut DB::QueryBuilder) -> $crate::query_builder::BuildQueryResult {
-                try!(out.push_identifier($($table)::*::name()));
+                try!($($table)::*.from_clause().to_sql(out));
```

Comment:
Changing from just pushing the table name to using `QuerySource` is a nice refactoring.